### PR TITLE
for-await-of: dstr-binding templates

### PR DESCRIPTION
**Pull request by rwaldron**
This is a work in progress, do not merge

Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: src/dstr-binding/default/for-await-of-const.template [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+               1. Assert: lhsKind is lexicalBinding.
+               2. Assert: lhs is a ForDeclaration.
+               3. Let status be the result of performing BindingInitialization
+                  for lhs passing nextValue and iterationEnv as arguments.
+          [...]
+---*/
+
+var iterCount = 0;
+
+async function fn() {
```
Comment (leobalter) [inline src/dstr-binding/default/for-await-of-const.template]: clone these templates to support async generators as well.

file names can be like:

- `src/dstr-binding/default/for-await-async-func-const.template`
- `src/dstr-binding/default/for-await-async-gen-const.template`

and the `path` (frontmatter) be like:

- `path: language/statements/for-await-async-function/dstr-const-`
- `path: language/statements/for-await-async-generator/dstr-const-`

**Comment by rwaldron** [comment]
After discussing offline, @leobalter agreed to let me do this: 

> clone these templates to support async generators as well.

...in a follow up PR

File: src/dstr-binding/default/for-await-of-const.template (hunk 1 above)
Comment (leobalter) [inline src/dstr-binding/default/for-await-of-const.template]: In a private chat, we decided to have this in a follow up PR, the tests are technical valid and the follow up fix will only add extra coverage.

**Comment by leobalter** [comment]
Did you sign the CLA?

**Comment by rwaldron** [comment]
🖕

**Pull request merged**